cmd/controller/ctrl.go

```diff
@@ -66,24 +71,28 @@ func NewController(
 	return controller
 }
 
-// getUid will return unique id for the disk block device
-func getUid(device *udev.Udevice) string {
-	return NDMPrefix +
-		util.Hash(device.PropertyValue(udev.UDEV_WWN)+
-			device.PropertyValue(udev.UDEV_MODEL)+
-			device.PropertyValue(udev.UDEV_SERIAL)+
-			device.PropertyValue(udev.UDEV_VENDOR))
-}
-
 // Run will discover the local disks and push them to the etcd
 func (c *Controller) Run(threadiness int, stopCh <-chan struct{}) error {
 	glog.Info("Started the controller")
 
-	err := pushDiskResources(c)
+	err := c.pushDiskResources()
 	if err != nil {
 		glog.Error("Unable to push disk objects to etcd : ", err)
 		return err
 	}
+	//This infinite loop will help to get event details send by udev-event module
+	//event details send using channel.
+	go func() {
+		processEvents := newProcessEvents(c)
```

Comment:
as monitor setup a blocking call that why we decided to run it in another go routine cc @pawanpraka1 . It can be moved out but latter if other event module or blocking call comes then it is good to run it in a routine.

Reply:
newProcessEvents() returns object of ProcessEvents struct so I made it a function.